File: go_version/models/player.go
package models

type Player struct {
	ID      int
	Name    string
	Score   int
	History map[int]int
}

func NewPlayer(id int, name string) Player {
	return Player{
		ID:      id,
		Name:    name,
		Score:   0,
		History: map[int]int{},
	}
}

func (p *Player) AddScores(score, round int) {
	p.History[round] = score
	p.Score += score
}

File: csharp_version/GinScoreKeeper/Models/Player.cs
using System.Collections.Generic;

namespace GinScoreKeeper.Models
{
    public class Player
    {
        public Player()
        {
            ScoreHistory = new Dictionary<int, int>();
        }

        public int Id { get; set; }
        public string Name { get; set; }
        public int Score { get; set; }
        public Dictionary<int, int> ScoreHistory { get; set; }
    }
}